homeassistant/components/dhcp/__init__.py

```diff
@@ -318,6 +319,12 @@ async def async_start(self):
 
     def handle_dhcp_packet(self, packet):
         """Process a dhcp packet."""
+        # Local import because importing from scapy has side effects such as opening
+        # sockets
+        from scapy.layers.dhcp import DHCP  # pylint: disable=import-outside-toplevel
+        from scapy.layers.inet import IP  # pylint: disable=import-outside-toplevel
+        from scapy.layers.l2 import Ether  # pylint: disable=import-outside-toplevel
```

Comment:
This can get called often, can we import these in `async_start` or some place else where it only gets called once?

https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Import_Statement_Overhead

Reply:
Interesting, I thought the import was only costly the first time around.
If imported in `async_start`, will the symbols be visible to other functions in the class?

Reply:
Or do you have something like this in mind?
```py
    def async_start(self):
        from scapy.layers.dhcp import DHCP
        self.scapy_DHCP = DHCP
```

Reply:
Maybe something like https://github.com/bdraco/home-assistant/commit/3051c1859d7d0e9b52986f0b15a4e807d758cdd8

Reply:
Imports are cached: https://docs.python.org/3/reference/import.html#the-module-cache

Reply:
Right, but don't we still have to pay the overhead to lookup the module in the cache?

I guess it probably doesn't matter enough here since we don't hit `handle_dhcp_packet` that the small amount of overhead will matter

Reply:
I think that it's just resolving import to current package name + a dict lookup:

```python
import sys
print(sys.modules)
```

Reply:
It's probably way faster now in python3. The last time I hit this problem was from python2 era 🙈

Reply:
btw I noticed that both `pr2modules` and `ruamel` load themselves by default. Opened PR to remove ruamel because it's unused #55672